libs/libc/machine/arch_atomic.c

```diff
@@ -34,7 +34,8 @@
 
 #define STORE(n, type)                                    \
                                                           \
-  void __atomic_store_ ## n (volatile void *ptr,          \
+  void weak_function __atomic_store_ ## n (               \
+                             volatile void *ptr,          \
```

Comment:
Can we add `FAR` to pointers? Or maybe in the next PR?

Reply:
@xiaoxiang781216 Done